### ISSUE TITLE
new data for J/GBrowse for WS274

### ISSUE THREAD

**Issue by epaule** [WormBase/website#7406]
for B.malayi:

 BLAT_Caen_Nanopore_BEST expressed_sequence_match
 BLAT_Caen_Nanopore_OTHER expressed_sequence_match

Nic Wheeler of the Zamanian lab provided us with IsoSeq data which was previously presented at the PacBio North America User Group meeting. We applied the PacBio postprocessing pipeline as described here https://github.com/PacificBiosciences/IsoSeq_SA3nUP/wiki and aligned the resulting sequences back to the genome using BLAT.

**Comment by epaule** [comment]
but there are also:
BLAT_IsoSeq_BEST	expressed_sequence_match
BLAT_IsoSeq_OTHER	expressed_sequence_match

which are (at least in the B.malayi case) from:
Nic Wheeler of the Zamanian lab provided us with IsoSeq data which was previously presented at the PacBio North America User Group meeting. We applied the PacBio postprocessing pipeline as described here https://github.com/PacificBiosciences/IsoSeq_SA3nUP/wiki and aligned them using BLAT.

**Comment by scottcain** [comment]
@epaule just to be clear: that description for IsoSeq is only for the B. malayi data? I can do that (have a different description between species) it just takes a little extra config work.

**Comment by epaule** [comment]
the description would be for B.malayi only, if you can do it.

I think T.muris has also BLAT_IsoSeq_BEST/OTHER data, which is from another source. but the descriptions are quite "sparse". I will ask @fayerodgers  if she can add some fluff to T.muris.

**Comment by fayerodgers** [comment]
I've put the T muris IsoSeq BAM here for you (I think putting it in your S3 bucket will be the safest place for it!): ftp://ftp.ebi.ac.uk/pub/databases/wormbase/STAFF/fr7/Tmuris_isoseq.bam

The track description for this BAM could be:
"PacBio IsoSeq (RSII) reads from adult male and adult female T muris libraries, generated by the Parasite Genomics group at the Wellcome Sanger Institute. Full-length non-chimeric reads were aligned with GMAP".

@epaule for the other track, rather than showing the collapsed and then realigned best/other BLATs, can we just show the collapsed alignments? Then we can still see what the transcripts were built from. The description for that could be:
"Aligned full-length non-chimeric reads from adult male and adult female T muris libraries, with similar alignments collapsed into one representative structure".

**Comment by epaule** [comment]
it depends on the goal of the exercise: if it is to show what created the WB transcripts, the BLAT tracks are handy. But if you want to show the best possible IsoSeq data, then the BAMs are way better and we could drop the BLAT tracks.

**Comment by scottcain** [comment]
Another thing I just want to double check on: the BLAT_IsoSeq_BEST/OTHER are different from the BLAT_Caen_IsoSeq_BEST/OTHER features, right? I'm reasonably sure that the BLAT_Caen_IsoSeq are elegans sequences blatted against a variety of assemblies, where as the BLAT_IsoSeq are "same against same" blats.  I didn't realize until today that they were different, when I was trying to diagnose why my pipeline didn't pickup the BLAT_IsoSeq features.

**Comment by epaule** [comment]
Yes, the ones with BLAT_Cean are all IsoSeq except the ones from the
respective species. So they should appear in basically any species that we
build since we added them (C.elegans would have the T.muris & B.malayi
IsoSeq sequences in these tracks).

**Comment by fayerodgers** [comment]
My beef with the BLAT tracks for T muris is that we corrected the aligned/collapsed sequences based on the genome, then BLAT-ed them. I think we only did that because it was the easiest way to get them into transcript builder, IIRC. So it renders labelling something a "best" hit kind of meaningless, because it will be a 100% match by definition of the way it was produced! So my vote would be to drop them :)

**Comment by scottcain** [comment]
All three tracks are currently on staging jbrowse (isoseq best, isoseq other and isoseq bam):

https://staging.wormbase.org/tools/genome/jbrowse-simple/?data=data%2Ft_muris_PRJEB126&loc=TMUE_LG2%3A10520140..10528900&tracks=Curated_Genes%2CIsoseq%20collection%20(best)%2CIsoSeq%20collection%20alignments%2CIsoseq%20collection%20(other)

It's unclear to me why the "other" track would be taller than the bam track, but I imagine that it's due to differences with how the similarities/query sequences were calculated/constructed.  Of course, please let me know if you'd like changes in the appearance (like different colors for the forward and reverse matches in the BAM track).

With regard to the muris and malayi IsoSeq tracks--it would be helpful for me for their sources to be different (like IsoSeq_Bmalayi_BEST/OTHER and IsoSeq_Tmuris_BEST/OTHER).  At the moment, my pipeline doesn't support the type:method combination being the same but the track config being different between assemblies.  While I could rework the pipeline, it would be a lot easier if the source were different instead.  I don't need it for 274, as I've done that portion of the build by hand already but it would be great if that could happen for 275.  Alternatively, if we don't expect the data in these tracks to change, I could just move the contents to the S3 bucket where "long term" data are stored and take the per-release processing for these data out of the pipeline all together.

**Comment by epaule** [comment]
I am for including all three tracks, as it looks like not all sequences from the BAM made it into the BLAT alignments (and the transcript builder).

The reason why the tracks are different height is because one uses the BAM rendering path and the other the generic canvas features (i assume). But I don't think it matters.

**Comment by fayerodgers** [comment]
Hi @scottcain, thanks! However...in light of all the above we (me, @epaule, @khowe and @gw3) discussed this today, and decided the time was nigh for a more general review of tracks on JBrowse. We've come up with the following suggestions together: 

- For all species, remove cross species nucleotide alignment tracks (in the “sequence similarity” -> “nucleotide” panel. These are:

1. “WormBase nematode mRNAs/EST (best)” and “WormBase nematode mRNAs/EST (other) ”
2. “non-C.elegans IsoSeq collection (best)” and "non-C.elegans IsoSeq collection (other)".  
3. For non-C.elegans species, “Nanopore matches (best)” and “Nanopore matches (other)”.  See below for suggestions for the C. elegans Nanopore track.

- For all species, remove all tracks which have the word “(other)” in the track label. 

- For all species, remove the word “(best)” from tracks with “(best)” in the track label. We will need to suggest alternative descriptions for these tracks (the description currently refers to the “other” tracks).

- For species with IsoSeq data (T muris and B malayi), move “IsoSeq collection alignments”  (the BAM track) to the “Expression” panel. The descriptions for these tracks will be different for each species (have already provided for T muris).

- For species with IsoSeq data, rename the “IsoSeq collection (best)” track to “Collapsed IsoSeq structures” and move it to the “Genes” -> “supporting evidence” panel. For T muris, the description of this track can be: "Aligned and filtered full-length non-chimeric IsoSeq reads with similar alignments collapsed into one representative structure". 

- Move the “Nanopore matches” track from the section “Sequence similarity”->”Nucleotide” to “Genes”->”Supporting evidence”. Reword the description of this track so that it doesn’t mention “This data represents the best alignment of Nanopore transcripts to the reference genome”.

- This should leave the section “Sequence similarity”->”Nucleotide” empty and it should be removed.

Does that all sound sensible?

**Comment by scottcain** [comment]
Yes, it does.  The "other" tracks have bugged me for a while.  Are the ones already under genes/supporting evidence on the chopping block too (that is, ESTs (other) and mRNA/ncRNA (other))?

**Comment by scottcain** [comment]
@fayerodgers , @gw3 , @khowe , @epaule 

I believe I made all of the changes suggested above, with the exception of my suggestion of simplifying the sequence similarity/protein category, but given that there were quite a few things touched (and I have a little bit of a cold, limiting my cognitive abilities at the moment), I'd appreciate it if you could take a look at the staging JBrowse to see if all looks as you would have thought:

https://staging.wormbase.org/tools/genome/jbrowse-simple/

**Comment by gw3** [comment]
Nanopore appears to be missing from "Genes"->"Supporting Evidence" in 
elegans N2.

**Comment by scottcain** [comment]
Got it in there now @gw3 . Thanks for pointing it out.

**Comment by fayerodgers** [comment]
Hey @scottcain, looks good. Just a couple of things. In the "supporting evidence" panel, could you remove these lines from the ESTs and mRNAs/ncRNAs track descriptions respectively:

- "Other EST matches, some of which may represent repetitive elements, are shown in the track labeled 'ESTs (other)'" 

- "Other cDNA matches, some of which may represent repetitive elements, are shown in the track labeled 'mRNAs/ncRNAs (other)'"

For T muris, could you also add this description to the Collapsed IsoSeq Structures track?: "Aligned and filtered full-length non-chimeric IsoSeq reads with similar alignments collapsed into one representative structure". Would the same description work for B malayi @epaule ?

Thanks!

**Comment by epaule** [comment]
yup It could work

**Comment by gw3** [comment]
Hi @scottcain, is the Nanopore track meant to be displaying data in JBrowse staging?

It is showing the track label, but no alignments.

**Comment by scottcain** [comment]
@fayerodgers I updated the track descriptions as you suggested.

@gw3 what, you want data too? Are you never satisfied? OK, fixed (and I think I fixed the pipeline to do it right the next time too.

Also, the Collapsed IsoSeq collection is in B malayi now.

**Comment by tharris** [comment]
@fayerodgers @gw3 Do you consider this resolved?
